runtime/lua/vim/keymap.lua

```diff
@@ -0,0 +1,223 @@
+local keymap = {}
+
+-- Have to use a global to handle re-requiring this file and losing all of the keymap.
+--  In the future, the C code will handle this.
```

Comment:
Is the idea that all of this PR will eventually be rewritten as C or C bindings?